### PR TITLE
JSON and Text as acceptable media type for eth2 sign operation

### PR DESCRIPTION
**Pull request by usmansaleem**
#281

### CODE REVIEW

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/signing/SigningAcceptanceTestBase.java [hunk 1]
```diff
@@ -37,7 +37,18 @@ protected void setupSigner(final String mode, final Map<String, String> env) {
...
-    response.then().contentType(ContentType.TEXT).statusCode(200);
-    return Bytes.fromHexString(response.body().print());
+    return verifyAndGetSignatureResponse(response, ContentType.TEXT);
+  }
+
+  protected Bytes verifyAndGetSignatureResponse(
+      final Response response, final ContentType contentType) {
+    response.then().contentType(contentType).statusCode(200);
+    final String signature;
+    if (contentType == ContentType.JSON) {
```
Comment (jframe) [inline acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/signing/SigningAcceptanceTestBase.java:47]: Doesn't look like this is actually use for the json content type. Would be good to have an AT that shows we return json. The ableToSignWithoutSigningRootField only checks the contentType is set correctly, it doesn't check the response.

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/signing/BlsSigningAcceptanceTest.java [hunk 2]
```diff
@@ -147,8 +148,12 @@ public void failsIfSigningRootDoesNotMatchSigningData() throws JsonProcessingExc
     assertThat(response.getStatusCode()).isEqualTo(500);
   }
 
-  @Test
-  public void ableToSignWithoutSigningRootField() throws JsonProcessingException {
+  @ParameterizedTest
+  @EnumSource(
```
Comment (jframe) [inline acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/signing/BlsSigningAcceptanceTest.java:152]: Would also be good to have this on a test that sends through the entire set of data, say the signDataWithKeyLoadedFromUnencryptedFile AT

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/eth2/Eth2SignForIdentifierHandler.java [hunk 3]
```diff
@@ -213,7 +216,19 @@ private UInt64 toUInt64(final tech.pegasys.teku.infrastructure.unsigned.UInt64 u
...
+    final String acceptableContentType = routingContext.getAcceptableContentType();
+    final String contentType;
+    final String body;
+
+    LOG.trace("Acceptable Content Type {}", acceptableContentType);
+    if (Objects.equal("application/json", acceptableContentType)) {
+      contentType = JSON_UTF_8;
+      body = new JsonObject().put("signature", signature).encode();
+    } else {
+      contentType = TEXT_PLAIN_UTF_8;
```
Comment (jframe) [inline core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/eth2/Eth2SignForIdentifierHandler.java:228]: Does this default to text/plain? If I send a request to Accept: */* will this return the plain text response?

Comment (usmansaleem) [reply core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/eth2/Eth2SignForIdentifierHandler.java:228]: For `Accept` header with values `*/*` and `application/json` the `routingContext.getAcceptableContentType();`  returns `application/json`. If Accept header is any thing other than */*, application/json and text/<?>, the openapi router factory kicks in and returns a 403 I think (invalid media type).

File: acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/signing/SigningAcceptanceTestBase.java [hunk 4]
```diff
@@ -37,7 +37,18 @@ protected void setupSigner(final String mode, final Map<String, String> env) {
...
-    response.then().contentType(ContentType.TEXT).statusCode(200);
-    return Bytes.fromHexString(response.body().print());
+    return verifyAndGetSignatureResponse(response, ContentType.TEXT);
+  }
+
+  protected Bytes verifyAndGetSignatureResponse(
+      final Response response, final ContentType expectedContentType) {
+    response.then().contentType(expectedContentType).statusCode(200);
+    final String signature;
+    if (expectedContentType == ContentType.JSON) {
```
Comment (jframe) [inline acceptance-tests/src/test/java/tech/pegasys/web3signer/tests/signing/SigningAcceptanceTestBase.java:47]: nit: a ternary would probably read nicer

**Review by jframe — APPROVED** [review]

**Pull request merged**